src/input/JoystickInputDriver.cc

```diff
@@ -64,7 +64,9 @@ bool JoystickInputDriver::open()
 {
     stopRequest = false;
 
-    fd = ::open("/dev/input/js0", O_RDONLY);
+    char* path=const_cast<char*> (("/dev/input/js" + nr).c_str());
```

Comment:
This confuses me. Are there implementations of `open()` that doesn't take a `const char *`?
As we use `boost::format` other places, you could do that here as well; smth. like this:
```
auto path = boost::format("/dev/input/js %d") % nr;
this->fd = ::open(path.c_str(), O_RDONLY);
```

Reply:
thank you for your suggestion.
changed to boost::format